go-controller/pkg/ovn/egressip.go

```diff
@@ -1662,39 +1701,75 @@ func (e *egressIPZoneController) getTransitIP(nodeName string, wantsIPv6 bool) (
 	return nodeTransitIP.IP.String(), nil
 }
 
-// createReroutePolicyOps creates an operation that does idempotent updates of the
-// LogicalRouterPolicy corresponding to the egressIP status item, according to the
-// following update procedure:
-// - if the LogicalRouterPolicy does not exist: it adds it by creating the
-// reference to it from ovn_cluster_router and specifying the array of nexthops
-// to equal [gatewayRouterIP]
-// - if the LogicalRouterPolicy does exist: it adds the gatewayRouterIP to the
-// array of nexthops
-// This function should be called with lock on nodeZoneState cache key status.Node
-func (e *egressIPZoneController) createReroutePolicyOps(ops []ovsdb.Operation, podIPNets []*net.IPNet, status egressipv1.EgressIPStatusItem, egressIPName string) ([]ovsdb.Operation, error) {
-	isEgressIPv6 := utilnet.IsIPv6String(status.EgressIP)
+func (e *egressIPZoneController) getNextHop(egressNodeName, egressIP, egressIPName string, isLocalZoneEgressNode, isOVNManaged bool) (string, error) {
 	var nextHopIP string
+
 	var err error
+	isEgressIPv6 := utilnet.IsIPv6String(egressIP)
 	// NOTE: No need to check if status.node exists or not in the cache, we are calling this function only if it
 	// is present in the nodeZoneState cache. Since we call it with lock on cache, we are safe here.
-	isLocalZoneEgressNode, _ := e.nodeZoneState.Load(status.Node)
 	if isLocalZoneEgressNode {
-		gatewayRouterIP, err := e.getGatewayRouterJoinIP(status.Node, isEgressIPv6)
-		if err != nil {
-			return nil, fmt.Errorf("unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %w", status.Node, isEgressIPv6, err)
+		if isOVNManaged {
+			gatewayRouterIP, err := e.getGatewayRouterJoinIP(egressNodeName, isEgressIPv6)
+			if err != nil {
+				return "", fmt.Errorf("unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %w",
+					egressNodeName, isEgressIPv6, err)
+			}
+			nextHopIP = gatewayRouterIP.String()
+		} else {
+			mgntPort := &nbdb.LogicalSwitchPort{Name: fmt.Sprintf("k8s-%s", egressNodeName)}
+			mgntPort, err := libovsdbops.GetLogicalSwitchPort(e.nbClient, mgntPort)
+			if err != nil && !apierrors.IsNotFound(err) {
+				return "", fmt.Errorf("failed to get next hop IP for non-ovn managed network and egress IP %s for node %s "+
+					"because unable to get management port: %v", egressIPName, egressNodeName, err)
+			} else if err != nil {
+				return "", nil
```

Comment:
NIT: Is it worth adding a log message saying that the node is not found?

Reply:
done